src/platform/windows/misc.cpp

```diff
@@ -454,74 +594,102 @@ namespace platf {
+      HANDLE user_token = retrieve_user_token(elevated);
+      if (!user_token) {
+        // Fail the launch rather than risking launching with Sunshine's permissions unmodified.
+        ec = std::make_error_code(std::errc::no_such_process);
         return bp::child();
       }
 
-      // Launch the process with the duplicated shell token.
-      // Set CREATE_BREAKAWAY_FROM_JOB to avoid the child being killed if SunshineSvc.exe is terminated.
-      // Set CREATE_NEW_CONSOLE to avoid writing stdout to Sunshine's log if 'file' is not specified.
-      ret = CreateProcessAsUserW(shell_token,
-        NULL,
-        (LPWSTR) wcmd.c_str(),
-        NULL,
-        NULL,
-        !!(startup_info.StartupInfo.dwFlags & STARTF_USESTDHANDLES),
-        EXTENDED_STARTUPINFO_PRESENT | CREATE_UNICODE_ENVIRONMENT | CREATE_NEW_CONSOLE | CREATE_BREAKAWAY_FROM_JOB,
-        env_block.data(),
-        start_dir.empty() ? NULL : start_dir.c_str(),
-        (LPSTARTUPINFOW) &startup_info,
-        &process_info);
+      // Use RAII to ensure the shell token is closed when we're done with it
+      auto token_close = util::fail_guard([user_token]() {
+        CloseHandle(user_token);
+      });
 
-      if (!ret) {
-        auto error = GetLastError();
-
-        if (error == 740) {
-          BOOST_LOG(info) << "Could not execute previous command because it required elevation. Running the command again with elevation, for security reasons this will prompt user interaction."sv;
-          startup_info.StartupInfo.wShowWindow = SW_HIDE;
-          startup_info.StartupInfo.dwFlags = startup_info.StartupInfo.dwFlags | STARTF_USESHOWWINDOW;
-          std::wstring elevated_command = L"tools\\elevator.exe ";
-          elevated_command += wcmd;
-
-          // For security reasons, Windows enforces that an application can have only one "interactive thread" responsible for processing user input and managing the user interface (UI).
-          // Since UAC prompts are interactive, we cannot have a UAC prompt while Sunshine is already running because it would block the thread.
-          // To work around this issue, we will launch a separate process that will elevate the command, which will prompt the user to confirm the elevation.
-          // This is our intended behavior: to require interaction before elevating the command.
-          ret = CreateProcessAsUserW(shell_token,
-            nullptr,
-            (LPWSTR) elevated_command.c_str(),
-            nullptr,
-            nullptr,
+      // Populate env with user-specific environment variables
+      if (!merge_user_environment_block(env, user_token)) {
+        ec = std::make_error_code(std::errc::not_enough_memory);
+        return bp::child();
+      }
+
+      // Create the STARTUPINFOEXW object for the new process
+
+      if (elevated && is_running_as_system()) {
+        // Impersonate the current user and launch the new process
+        ec = impersonate_current_user(user_token, [&]() {
```

Comment:
We only need this single codepath, not the `!is_token_same_user_as_process(user_token)` one below. You won't need to check `elevated` because `retrieve_user_token()` already grabbed the correct token for you, and you don't have to check `is_running_as_system()` since you're already inside an `if (is_running_as_system())` codepath.